pkg/storaged/lvol-tabs.jsx

```diff
@@ -250,19 +267,84 @@ function lvol_grow(client, lvol, info, to_fit) {
         return;
     }
 
+    const pvs_as_spaces = pvs_to_spaces(client, client.vgroups_pvols[vgroup.path].filter(pvol => pvol.FreeSize > 0));
+
+    let initial_pvs;
+
+    if (!has_subvols)
+        initial_pvs = pvs_as_spaces;
+    else {
+        initial_pvs = [];
+
+        // Step 1: Find the spaces that are already used for a
+        // subvolume.  If a subvolume uses more than one, prefer the
+        // one with more available space.
+        for (const sv of subvols) {
+            let sel = null;
+            for (const p in sv) {
+                for (const spc of pvs_as_spaces)
+                    if (spc.block.path == p && (!sel || sel.size < spc.size))
+                        sel = spc;
+            }
+            if (sel)
+                initial_pvs.push(sel);
+        }
+
+        // Step 2: Select missing one randomly.
+        for (const pv of pvs_as_spaces) {
+            if (initial_pvs.indexOf(pv) == -1 && initial_pvs.length < subvols.length)
+                initial_pvs.push(pv);
+        }
+    }
+
+    function max_size(spaces) {
+        const layout = lvol.Layout;
+        const pvs = spaces.map(s => s.pvol);
+        const n_pvs = pvs.length;
+        const sum = pvs.reduce((sum, pv) => sum + pv.FreeSize, 0);
+        const min = Math.min.apply(null, pvs.map(pv => pv.FreeSize));
+
+        if (!has_subvols) {
+            return sum;
+        } else if (layout == "raid0") {
+            return n_pvs * min;
+        } else if (layout == "raid1") {
+            return min;
+        } else if (layout == "raid10") {
+            return (n_pvs / 2) * min;
+        } else if ((layout == "raid4" || layout == "raid5")) {
+            return (n_pvs - 1) * min;
+        } else if (layout == "raid6") {
+            return (n_pvs - 2) * min;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230918-080053-f89e42e4-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)